packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts

```diff
@@ -862,16 +861,19 @@ export async function downloadResultArchive(
             {
                 exportId: jobId,
                 exportIntent: ExportIntent.TRANSFORMATION,
+                ...(downloadArtifactId !== undefined && {
+                    exportContext: { transformationExportContext: { downloadArtifactId, downloadArtifactType } },
+                }),
             },
             pathToArchive
         )
     } catch (e: any) {
-        downloadErrorMessage = (e as Error).message
+        const downloadErrorMessage = (e as Error).message
         getLogger().error(`CodeTransformation: ExportResultArchive error = ${downloadErrorMessage}`)
         telemetry.codeTransform_logApiError.emit({
             codeTransformApiNames: 'ExportResultArchive',
             codeTransformSessionId: CodeTransformTelemetryState.instance.getSessionId(),
-            codeTransformJobId: transformByQState.getJobId(),
+            codeTransformJobId: jobId,
             codeTransformApiErrorMessage: downloadErrorMessage,
             codeTransformRequestId: e.requestId ?? '',
             result: MetadataResult.Fail,
```

Comment:
Pretty sure this will cause conflicts with this one: https://github.com/aws/aws-toolkit-vscode/pull/5044

Reply:
I left a comment on your PR as to what can be changed there to make essentially the same change I'm doing here -- otherwise I'm fine with this block of code being completely overwritten by your change when it comes time to merge

Reply:
Yeah I made the change your requested! Tbh we should never use globals that is not constants imo. And, if strictly required limit them to be project specific / internal to the class they are used in to scope them down. Great you are driving this.